support weights.column in GLM type models

kevinykuo opened the pull request:
Closes #217

kevinushey commented on R/ml_generalized_linear_regression.R:
```diff
@@ -27,6 +28,7 @@ ml_generalized_linear_regression <-
            features,
            intercept = TRUE,
            family = gaussian(link = "identity"),
+           weights_column = NULL,
```
Given that other parameters here use `dot.case` rather than `snake_case`, should this be `weights.column`?

kevinushey commented on R/ml_generalized_linear_regression.R:
```diff
@@ -88,6 +91,10 @@ ml_generalized_linear_regression <-
 
+  if (!is.null(weights_column))
+    model <- model %>%
+    invoke("setWeightCol", weights_column)
```
The indentation here is a bit misleading -- can we just wrap the expression in braces and indent continuations for the pipe operator?

kevinushey commented:
LGTM aside from some nitpicky stylistic concerns.

kevinykuo commented:
@kevinushey addressed comments; also corrected the regularization parameter in tests.

Could we have a feature request in the IDE to indent pipe continuations in `if` blocks with a single statement? :P
